src/classes/SRVC_UniqueRecord.cls

```diff
@@ -0,0 +1,159 @@
+
+        for (Sobject record: sObjectRecords) {
+
+            if (!idToSObjectRecordMap.containsKey(record.Id)) {
+                idToSObjectRecordMap.put(record.Id, record.getSObjectType().newSobject(record.Id));
+            }
+
+            SObject uniqueRecord = this.combineRecords(record, idToSObjectRecordMap.get(record.id)); 
+
+            idToSObjectRecordMap.put(record.Id, uniqueRecord);
+        }
+
+        return idToSObjectRecordMap.values();
+    }
+
+    /******************************************************************************************************
+    @description Method to combine recurring occurance of an SObject Records
+    @params SObjectRecord The SObjectRecord to combine the field for
+    @params newSobjectRecord A new instance of the SOBject record
+    @params allChildRelationshipNames A List of all ChildRelationshipNames for the given SObjectType
+    @return interimRecord An interming record to hold the combined record
+    ********************************************************************************************************/
+    @TestVisible
+    private SObject combineRecords(SObject sObjectRecord, SObject newSobjectRecord) {
+
+        List<String> childRelationshipNames = UTIL_Describe.getAllChildRelationshipNames(sObjectRecord.getSObjectType());
+
+        Map<String, Object> fieldNameToValues = sObjectRecord.getPopulatedFieldsAsMap();
+        SObject interimRecord = newSobjectRecord;
+        //run code in loop below to combine records together
+        for (String fieldName: fieldNameToValues.keySet()) {
+            if (!childRelationshipNames.contains(fieldName)) {  // Do not add field Names from related records
+                interimRecord.put(fieldName, fieldNameToValues.get(fieldName));
+            }
+        }
+        return interimRecord;
+
+    }
+
+    /******************************************************************************************************
+    @description Filters SObjects into two lists, one with ids, one without ids
+    @params allSObjectRecords The list of records to be checked and made unique
+    @return returnMap A key value pair for recordsWithoutIds and recordsWithIds
+    ********************************************************************************************************/
+    @TestVisible
+    private Map<String, List<SObject>> filterRecordsForIds(List<SObject> allSObjectRecords) {
+        Map<String, List<SObject>> returnMap = new Map<String, List<SObject>>();
+
+        List<SObject> recordsWithoutIds = new List<SObject>();
+        List<SObject> recordsWithIds = new List<SObject>();
+
+        for (SObject sObjectToCheck: allSObjectRecords) {
+            if (sObjectToCheck.Id == NULL) {
+                recordsWithoutIds.add(sObjectToCheck);
+            }
+            else {
+                recordsWithIds.add(sObjectToCheck);
+            }
+        }
+
+        returnMap.put('withoutIds', recordsWithoutIds);
```

Comment:
On the picky side, I noticed this during our demo yesterday: shall we make these keys Constants? Also, maybe we can capitalize them?

Reply:
Making constants will require additional variable declaration, weren't we trying to avoid such scenarios?

Reply:
If we are going that route, I am more comfortable making an inner Enum class and using that.

Reply:
Will update the code accordingly.

Reply:
@xinnanlisf @MatthewBlanski thanks for the suggestions. I have updated the code accordingly can you please give it another review? TIA.